bin/systemml-standalone.py

```diff
@@ -0,0 +1,200 @@
+if sys.platform == 'cygwin':
+    classpath_sep = os.pathsep
+    classpath_sep = ';'
+    lib_dir = os.popen('cygpath -pw ' + lib_dir).read().strip()
+    lib_dir_star = '"' + lib_dir + "\*" + '"'
+    systemml_jar = '"' + os.popen('cygpath -pw ' + systemml_jar).read().strip() + '"'
+    hadoop_home = '"' + os.popen('cygpath -pw ' + hadoop_home).read().strip() + '"'
+    log4j_properties_path = '"' + os.popen('cygpath -pw ' + log4j_properties_path).read().strip() + '"'
+    user_dir = '"' + os.popen('cygpath -pw ' + user_dir).read().strip() + '"'
+    script_file = '"' + os.popen('cygpath -pw ' + script_file).read().strip() + '"'
+    systemml_config_path = '"' + os.popen('cygpath -pw ' + systemml_config_path).read().strip() + '"'
+    classpath = lib_dir_star + '\\' + classpath_sep + systemml_jar
+else:
+    #classpath = '"' + lib_dir_star + '"' + os.pathsep + '"' + systemml_jar + '"'
+    classpath = lib_dir_star + os.pathsep + systemml_jar
+
+
+# Set the HADOOP_HOME environment variable
+if 'HADOOP_HOME' not in os.environ:
+    os.environ['HADOOP_HOME'] = hadoop_home
+
+
+print('================================================================================')
+
+# Set default Java options
+systemml_default_java_opts = \
+    '-Xmx8g -Xms4g -Xmn1g ' + \
+    '-cp ' + classpath + ' ' + \
+    '-Dlog4j.configuration=file:' + log4j_properties_path + ' ' \
+    '-Duser.dir=' + user_dir
+
+
+# Reads in key-value pairs from the conf/systemml-env.sh file
+def parse_env_file(env_file_path):
+    env_vars = {}
+    with open(env_file_path) as f:
+        for l in f:
+            l = l.strip()
+            if l and not(l.startswith('#')) and '=' in l:
+                k, v = l.split('=', 1)
+                k = k.strip()
+                v = v.strip()
+                if len(v) > 0:
+                    quoted = v[0] == v[len(v) - 1] and v[0] in ['"', "'"]
+
+                # strip quotes
+                if quoted:
+                    v = v[1:-1]
+                env_vars[k] = v
+    return env_vars
+
+
+# Add any custom Java options set by the user at command line, overriding defaults as necessary.
+if 'SYSTEMML_JAVA_OPTS' in os.environ:
+    systemml_java_opts = os.environ['SYSTEMML_JAVA_OPTS']
+    systemml_default_java_opts = systemml_default_java_opts + ' ' + systemml_java_opts
+    # del os.environ['SYSTEMML_JAVA_OPTS']
+
+# Add any custom Java options set by the user in the environment variables file,
+# overriding defaults as necessary.
+systemml_env_path = join(project_root_dir, 'conf', 'systemml-env.sh')
+if exists(systemml_env_path):
+    env_vars = parse_env_file(systemml_env_path)
+    os.environ.update(env_vars)
+
+
+# Invoke the jar with options and arguments
+cmd = ['java', systemml_default_java_opts, 'org.apache.sysml.api.DMLScript', '-f', script_file, '-exec singlenode', '-config', systemml_config_path] + sys.argv[2:]
+# For debugging
+# print(' '.join(cmd))
+
+return_code = os.system(' '.join(cmd))
```

Comment:
I think the use of the newer `subprocess.call` function is encouraged now (actually, there is a `subprocess.run` function, but it is for >= 3.5): https://docs.python.org/3.6/library/subprocess.html#replacing-os-system